### PR TITLE
Add is_ocaml_gui_yet page 

### PR DESCRIPTION
**Pull request by IdaraNabuk**
The page is_ocaml_gui_yet will be added to ocaml.org

### CODE REVIEW

**Comment by sabine** [comment]
You have another PR for this at https://github.com/ocaml/ocaml.org/pull/1864. Could you close the one of the two that is no longer needed please?

**Comment by IdaraNabuk** [comment]
I'm sorry I forgot to close the other one. I've done that now.

File: data/is_ocaml_yet/gui.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+OCaml, traditionally known for its strength in systems programming, formal verification, and as the language of choice for numerous academic endeavors, is steadily maturing in the GUI development landscape.
+
+With its strong static typing, emphasis on immutability, and excellent performance, it is gradually making a case for itself as a viable alternative to mainstream GUI development languages.
+
+OCaml has libraries that aid an easy integration and usage of Graphical User Interface (GUI). It offers a range of GUI libraries, each with unique features and capabilities. These libraries allow developers to create user interfaces for their applications
+
+## Want to Help?
+
+Check out our GitHub repo and contribute there.
```
Comment (christinerose) [inline data/is_ocaml_yet/gui.md]: Maybe a link to the repo here?

File: data/is_ocaml_yet/gui.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    status: 🟡
+    description: |
+      These are libraries written entirely in OCaml. It has packages for syndicating content and working with RSS feeds.
+    packages:
+      - name: bogue
+      - name: lablgtk
+      - name: lablqml
+      - name: tsdl
+      - name: sfml
+  - name: 3d graphics
```
Comment (christinerose) [inline data/is_ocaml_yet/gui.md:16]: Do these names show up as titles or headers?

Comment (IdaraNabuk) [reply data/is_ocaml_yet/gui.md:16]: Display
This is how the names are displayed.

File: data/is_ocaml_yet/gui.md (hunk 1 above)
Comment (IdaraNabuk) [reply data/is_ocaml_yet/gui.md]: I'm not sure which repo to link to. The text is just a placeholder for now.

File: data/is_ocaml_yet/gui.md (hunk 2 above)
Comment (christinerose) [reply data/is_ocaml_yet/gui.md:16]: Thank you!

File: node_modules/.lsp/debug.log [hunk 3]
```diff
@@ -0,0 +1,642 @@
+Hello - from /home/richie/.vscode/extensions/jaredly.reason-vscode-1.7.13/bin.native.linux
```
Comment (gpetiot) [inline node_modules/.lsp/debug.log]: This file should be removed.

File: data/is_ocaml_yet/gui.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+  - name: image processing
+    status: 🟡
+    description: |
+      These ibraries provide functionalities to manipulate and process images. offer a range of functionalities for image processing tasks, from basic image manipulation to more complex operations. They make OCaml a powerful language for image processing.
+    packages:
+      - name: tsdl-image
+      - name: bimage
+      - name: ocamlsdl2-image
+  - name: Plotting
+    status: 🟡
```
Comment (gpetiot) [inline data/is_ocaml_yet/gui.md]: Maybe this one should be green

File: data/is_ocaml_yet/gui.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+      - name: async_graphics
+      - name: cairo2
+  - name: Web (Js bindings)
+    status: 🟡
+    description: |
+      These libraries are used to create bindings between OCaml and JavaScript, allowing OCaml code to interact with JavaScript libraries. They help in creating web applications using OCaml by providing a bridge between OCaml and JavaScript.
+    packages:
+      - name: chartjs
+  - name: image processing
+    status: 🟡
```
Comment (gpetiot) [inline data/is_ocaml_yet/gui.md]: Maybe this one should be green

File: data/is_ocaml_yet/gui.md [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+      - name: telemetry
+      - name: note
+      - name: iomux
+---
+
+OCaml, traditionally known for its strength in systems programming, formal verification, and as the language of choice for numerous academic endeavors, is steadily maturing in the GUI development landscape.
+
+With its strong static typing, emphasis on immutability, and excellent performance, it is gradually making a case for itself as a viable alternative to mainstream GUI development languages.
+
+OCaml has libraries that aid an easy integration and usage of Graphical User Interface (GUI). It offers a range of GUI libraries, each with unique features and capabilities. These libraries allow developers to create user interfaces for their applications
```
Comment (gpetiot) [inline data/is_ocaml_yet/gui.md]: Missing period at the end.

File: data/is_ocaml_yet/gui.md [hunk 7]
```diff
@@ -0,0 +1,74 @@
+---
+id: gui
+question: Is OCaml GUI Yet?
+answer: Not quite yet, but we're getting there!
+categories:
+  - name: Frameworks
+    status: 🟡
+    description: |
+      These are libraries written entirely in OCaml. It has packages for syndicating content and working with RSS feeds.
```
Comment (sabine) [inline data/is_ocaml_yet/gui.md]:
```suggestion
      These are libraries written entirely in OCaml. It has packages for syndicating content and working with RSS feeds.
```

RSS feed syndication doesn't make sense here, and yes, the libraries are written entirely in OCaml but why would I care? I mean what's this section about?

File: data/is_ocaml_yet/gui.md [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+      - name: vg
+      - name: gg
+      - name: async_graphics
+      - name: cairo2
+  - name: Web (Js bindings)
+    status: 🟡
+    description: |
+      These libraries are used to create bindings between OCaml and JavaScript, allowing OCaml code to interact with JavaScript libraries. They help in creating web applications using OCaml by providing a bridge between OCaml and JavaScript.
+    packages:
+      - name: chartjs
```
Comment (sabine) [inline data/is_ocaml_yet/gui.md]: Doesn't this actually belong on the "Is OCaml Web Yet?" page?

File: data/is_ocaml_yet/gui.md [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+  - name: image processing
+    status: 🟢
+    description: |
+      These libraries provide functionalities to manipulate and process images. They offer a range of functionalities for image processing tasks, like basic image manipulation to more complex operations. They make OCaml a powerful language for image processing.
```
Comment (sabine) [inline data/is_ocaml_yet/gui.md]: repeats "functionalities", very verbose without saying a lot

File: data/is_ocaml_yet/gui.md [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+      - name: ocamlsdl2-image
+  - name: Plotting
+    status: 🟢
+    description: |
+      These libraries are used to create visual representations of data. They provide functionalities to generate various types of plots, charts, and diagrams.
+    packages:
+      - name: plot
+      - name: owl-plot
+      - name: oplot
+      - name: plotty
```
Comment (sabine) [inline data/is_ocaml_yet/gui.md]: `owl-plot` and `plotty` don't seem to be on opam-repository, so they would need to be linked as github links (see `web.md` document).

**Review by sabine — COMMENTED** [review]
Here's a few comments. I also pushed a commit that renders the link to the page in the sidebar.

**Review by christinerose — APPROVED** [review]

File: data/is_ocaml_yet/gui.md (hunk 7 above)
Comment (gpetiot) [reply data/is_ocaml_yet/gui.md]: I think this is is something that dodged the proof reading, it came from the original copy-past from https://ocaml.org/docs/is-ocaml-web-yet#syndicationrss. @IdaraNabuk how about:
```suggestion
      These libraries provide a rich toolkit to develop GUI applications.
```
?

File: data/is_ocaml_yet/gui.md (hunk 8 above)
Comment (gpetiot) [reply data/is_ocaml_yet/gui.md]: Agreed, let's remove this category.

File: data/is_ocaml_yet/gui.md (hunk 10 above)
Comment (gpetiot) [reply data/is_ocaml_yet/gui.md]: Agreed, @IdaraNabuk the syntax is this one: https://github.com/ocaml/ocaml.org/blob/main/data/is_ocaml_yet/web.md?plain=1#L28-L31

File: data/is_ocaml_yet/gui.md (hunk 9 above)
Comment (gpetiot) [reply data/is_ocaml_yet/gui.md]: @IdaraNabuk I agree with Sabine, some descriptions from https://ocaml.org/docs/is-ocaml-web-yet are very direct:
- HTTP client libraries for making HTTP requests.
- Packages for generating static websites.
- Packages for creating mobile applications

There is no need to be more complicated than that.

```suggestion
      Packages for image-processing.
```

**Comment by IdaraNabuk** [comment]
Okay, thanks. I'll update the PR.

**Review by christinerose — APPROVED** [review]

File: data/is_ocaml_yet/gui.md (hunk 10 above)
Comment (gpetiot) [reply data/is_ocaml_yet/gui.md]: I just checked, I don't know why I've listed `plotty` either, I couldn't find it anywhere, let's just remove it from the list!
And `owl-plot` should be `owl-plplot`, but by checking the owl repository I'm not sure this package should be listed either (looks like something internal that got published on opam by mistake). So let's remove them both. If people want to add more libraries to the list in the future they can open a PR.

**Review by sabine — APPROVED** [review]
Thanks @IdaraNabuk, merging this now! :sparkles:

**Pull request merged**